test/ha/autoscaler_test.go

```diff
@@ -89,8 +89,26 @@ func TestAutoscalerHA(t *testing.T) {
 		t.Fatal("Failed to find new leader:", err)
 	}
 
-	t.Log("Verifying the old revision can still be scaled from 0 to some number after leader change")
-	e2e.AssertAutoscaleUpToNumPods(ctx, 0, 3, time.After(60*time.Second), true /* quick */)
+	t.Log("Verifying the old revision can still be scaled from 0 after leader change")
+	// Use WaitForEndpointState for scaling from 0. This is because WaitForEndpointState uses a much larger request
+	// timeout value than the one used by AssertAutoscaleUpToNumPods. AssertAutoscaleUpToNumPods uses the default
+	// vegeta request timeout value, which could result in non-200 responses when scaling from 0.
```

Comment:
That is not true IIRC. I've set it to `0` explicitly, which should mean that it actually has no timeout at all.

Reply:
The following are the logs:

```
    autoscale.go:160: Status = 0, want: 200
    autoscale.go:161: URL: http://35.238.72.232:80?sleep=500 Duration: 5.230915ms Body:
    autoscale.go:326: revision "autoscaler-h-a-nrzcyvvw-00001" #replicas: 0, want at least: -1
    autoscale.go:160: Status = 0, want: 200
    autoscale.go:161: URL: http://35.238.72.232:80?sleep=500 Duration: 5.241076ms Body:
```

What could cause these non-200 responses?

Reply:
I'd bet these are connection errors of some sorts. Note that they are returning in ~5ms so timeouts don't really make sense there.

Reply:
Probably need to print `res.Error` as well.